Automatización de salida gráfica de consulta - Pasos 12 al 14

ajcappo opened the pull request:
Se modificó el código para automatizar los puntos 12, 13 y 14 que figuran en:
https://github.com/Proyecto-Mandarina/salidagrafica-atlas/blob/master/salida-gr%C3%A1fica%20atlas%20instructivo.pdf

ajcappo committed:
Automatización de salida gráfica de consulta 

Se modificó el código para automatizar los puntos 12, 13 y 14 que figuran en:
https://github.com/Proyecto-Mandarina/salidagrafica-atlas/blob/master/salida-gr%C3%A1fica%20atlas%20instructivo.pdf

ajcappo committed:
Relación etiquetas_segmentos con colores

Estos son los pasos para poder trabajar en excel la información que contiene el archivo  ./estilo_segmento/segmento.qml

ajcappo committed:
Etiquetas_segmentos - colores

Archivo .csv en el que figuran las etiquetas de segmentos y los colores asociados a las mismas. 
También se incluye el campo de alerta para aquellos valores que se encuentran duplicados

ajcappo committed:
Update capaconsulta.qml

hernan-alperin reviewed (changes requested):
podés procesar el csv para que queda más prolijo?
o mejor aún, hacer un sql para
crear una tabla
colores_segmentos
con campos
segmento, color,  line_color, outline_color
?

ajcappo committed:
Archivo SQL - segmentos-colores

Archivo SQL con los colores asignados a cada segmento. 
Se eliminaron los valores duplicados

hernan-alperin reviewed (changes requested):
está bien. pero fijate que name debe llamarse segmento debe ser clave única, vos introducís 1 registro por cada linea del csv cuando lo ideal es sumarizarlas.

¿podés escribir un SQL que haga eso?
segmento, color, line_color, outline_color
con segmento sin repetirse y con ningún campo en ''

para ello quizás te sirva usar una postgres DB propia

ajcappo commented:
> está bien. pero fijate que name debe llamarse segmento debe ser clave única, vos introducís 1 registro por cada linea del csv cuando lo ideal es sumarizarlas.
> 
> ¿podés escribir un SQL que haga eso? segmento, color, line_color, outline_color con segmento sin repetirse y con ningún campo en ''
> 
> para ello quizás te sirva usar una postgres DB propia

Cambiar el name por segmento si, pero para unificar cada segmento con los estilos de colores en algunos casos va a haber problema, porque algunos segmentos tienen 2 estilos de algún color ( color, line_color, outline_color).
Por ejemplo el segmento 10:

('10', '', '', '172,54,240,0'),
('10', '', '', '255,255,255,255'),
('10', '', '172,54,240,255', ''),
('10', '', '255,0,0,255', ''),
('10', '172,54,240,255', '', ''),
con 2 outile_color, 2 color y 1 line_color

hernan-alperin commented:
corregir redundancias como las del segmento 10, etc
quedarse con el último distinto de ''
